services/nyx/patches/hongfuzz-2b-chunked.diff

```diff
@@ -0,0 +1,190 @@
++    return buf_size;
++  }
++
++  if (add_buf) {
++    read_fuzz_packets(add_buf, add_buf_size, packet_out_add, &packet_size_add, 64);
++  }
++
++  int num_mutations = rand() % 5 + 1;
++
++
++  for (int i = 0; i < num_mutations; ++i) {
++    if (add_buf && packet_size_add > 0 && (rand() % 2)) {
++      // Splice one index
++      packet_out[rand() % packet_size].mut = 1;
++    } else {
++      // Mutate one index
++      packet_out[rand() % packet_size].mut = 2;
++    }
++  }
++
++  size_t written = 0;
++
++  *out_buf = data->mutator_buf;
++  uint8_t* cur = data->mutator_buf;
++
++  for (size_t idx = 0; idx < packet_size; ++idx) {
++    chunk_size* size_out = (chunk_size*)cur;
++
++    if (packet_out[idx].mut == 1) {
++      // Splice
++      size_t splice_idx = rand() % packet_size_add;
++
++      if (written + sizeof(chunk_size) + packet_out_add[splice_idx].size >= max_size) {
++        return written;
++      }
++
++      memcpy(cur + sizeof(chunk_size), packet_out_add[splice_idx].buf, packet_out_add[splice_idx].size);
++      *size_out = packet_out_add[splice_idx].size;
++    } else {
++      if (written + sizeof(chunk_size) + packet_out[idx].size >= max_size) {
++        return written;
++      }
++
++      memcpy(cur + sizeof(chunk_size), packet_out[idx].buf, packet_out[idx].size);
++      if (packet_out[idx].mut == 2) {
++        // Mutate
++        run.dynfile->data = data->mutator_buf + sizeof(chunk_size);
++        run.dynfile->size = packet_out[idx].size;
++
++        queue_input = run.dynfile->data;
++        queue_input_size = run.dynfile->size;
++
++	run.global->mutate.maxInputSz = MAX_FILE - written - sizeof(chunk_size);
++
++	mangle_mangleContent(&run, NUMBER_OF_MUTATIONS);
++
++	// Truncate output
++	if (run.dynfile->size > chunk_size_mask) {
++		run.dynfile->size = chunk_size_mask;
++	}
++
++	if (run.dynfile->data != data->mutator_buf + sizeof(chunk_size)) {
++		abort();
++	}
++
++        packet_out[idx].size = run.dynfile->size;
++      }
++
++      *size_out = packet_out[idx].size;
++    }
++
++    cur += *size_out + sizeof(chunk_size);
++    written += *size_out + sizeof(chunk_size);
++  }
++
++  /* return size of mutated data */
++  return written;
++}
++
++#else
++
+ size_t afl_custom_fuzz(my_mutator_t *data, uint8_t *buf, size_t buf_size,
+                        u8 **out_buf, uint8_t *add_buf, size_t add_buf_size,
+                        size_t max_size) {
+@@ -132,6 +268,8 @@ size_t afl_custom_fuzz(my_mutator_t *data, uint8_t *buf, size_t buf_size,
+
+ }
+
++#endif
++
+ /**
+  * Deinitialize everything
+  *
+diff --git a/include/config.h b/include/config.h
+index d8177a75..6a588810 100644
+--- a/include/config.h
++++ b/include/config.h
+@@ -459,7 +459,7 @@
+    problems with complex programs). You need to recompile the target binary
+    after changing this - otherwise, SEGVs may ensue. */
+
+-#define MAP_SIZE_POW2 16
++#define MAP_SIZE_POW2 23
+
+ /* Do not change this unless you really know what you are doing. */
+
```

Comment:
This chunk is unrelated to the honggfuzz mutator

Reply:
Moved to increase-map-size.diff.